components/normalizer/src/u24.rs

```diff
@@ -55,6 +56,7 @@ impl From<U24> for u32 {
 }
 
 /// Conversion input out of `U24` range
+#[allow(clippy::exhaustive_structs)] // stable
 pub struct U24Error;
```

Comment:
Suggestion: We don't have these zero-sized error types much in ICU4X. It would be more consistent with ICU4X style to instead have an error enum, like

```rust
#[derive(Debug, Copy, Clone, PartialEq, Eq)]
#[non_exhaustive]
pub struct U24Error {
    /// Value is out of range of a U24
    Limit
}
```